src/twisted/internet/test/test_tcp.py

```diff
@@ -988,6 +1099,24 @@ def getListeningPort(self, reactor, factory, port=0, interface=''):
         return reactor.listenTCP(port, factory, interface=interface)
 
 
+    def connectToListener(self, reactor, address, factory):
+        """
+        Connect to the given listening TCP port.
+
+        @param reactor: The reactor under test.
+        @type reactor: L{IReactorTCP}
+
+        @param address: The listening's address.
+        @type address: L{IPv4Address} or L{IPv6Address}
+
+        @param factory: The client factory.
+        @type factory: L{ClientFactory}
+
+        @return: The connector
+        """
+        return reactor.connectTCP('127.0.0.1', address.port, factory)
```

Comment:
> I suppose this hasn't come up before because the existing tests use the socket module instead of Twisted's TCP client API (so they just have a bunch of semi-duplicated socket-module code). That makes me think there may have been some explicit decision not to use Twisted's client APIs to test these Twisted server-side behaviors.

[It seems like other tests used `socket`s directly to minimize the system under test](https://github.com/twisted/twisted/blob/de597aa01ea3e0db7ff95f31b0fecd74318fd285/src/twisted/internet/test/test_tcp.py#L1096-L1131).  I'll admit to using `connectTCP` here out of laziness: it allows me to use `MyClientFactory` which conveniently fires a `Deferred` on `connectionLost`.

> Also, if it makes sense to keep this method, it seems like the address parameter is a bit misleading since the connection is always to localhost.

[This is an unfortunate bit of platform specificity](https://github.com/twisted/twisted/pull/996/commits/94113c4cf9709f479e2e8f5d458da72066384487).  The other option i saw was to change `getListeningPort`'s default interface to `'127.0.0.1'`, which would cause `port.getHost().address` to be connectable on Windows. This seems like the right thing to do since I don't think tests should listen on my public IP address anyway, but I didn't want change an implicit interface that other tests might rely on.

Reply:
Addressed in: https://github.com/twisted/twisted/pull/996/commits/963e05a2ef24c11b5d78cc385868f0d85602aaac